analyzers/src/SonarAnalyzer.CSharp/Rules/RedundancyInConstructorDestructorDeclaration.cs

```diff
@@ -91,30 +90,31 @@ private static void CheckConstructorDeclaration(SonarSyntaxNodeReportingContext
 
         private static bool IsConstructorRedundant(ConstructorDeclarationSyntax constructorDeclaration, SemanticModel semanticModel) =>
             constructorDeclaration is { ParameterList.Parameters.Count: 0, Body.Statements.Count: 0 }
+            && !IsStructWithInitializedFieldOrProperty((BaseTypeDeclarationSyntax)constructorDeclaration.Parent, semanticModel)
             && (IsSinglePublicConstructor(constructorDeclaration, semanticModel)
                 || constructorDeclaration.Modifiers.Any(SyntaxKind.StaticKeyword));
 
+        private static bool IsStructWithInitializedFieldOrProperty(BaseTypeDeclarationSyntax parentType, SemanticModel semanticModel) =>
```

Comment:
`parentType` might be confused with nested types. I would rename the parameter.
```suggestion
        private static bool IsStructWithInitializedFieldOrProperty(BaseTypeDeclarationSyntax typeDeclaration, SemanticModel semanticModel) =>
```